lib/wallet/src/Cardano/Wallet/DB/Store/Transactions/TransactionInfo.hs

```diff
@@ -0,0 +1,220 @@
+        , WT.txInfoScriptValidity = txMetaScriptValidity <&> \case
+              False -> WT.TxScriptInvalid
+              True -> WT.TxScriptValid
+        }
+  where
+    tipH = getQuantity $ tip ^. #blockHeight
+    mkTxIn tx =
+        ( WT.TxIn
+          { inputId = getTxId (txInputSourceTxId tx)
+          , inputIx = txInputSourceIndex tx
+          }
+        , lookupTxOut (mkTxOutKey tx) decor
+        )
+    mkTxCollateral tx =
+        ( WT.TxIn
+          { inputId = getTxId (txCollateralSourceTxId tx)
+          , inputIx = txCollateralSourceIndex tx
+          }
+        , lookupTxOut (mkTxOutKeyCollateral tx) decor
+        )
+    mkTxWithdrawal w = (txWithdrawalAccount w, txWithdrawalAmount w)
+
+    mkTxCBOR :: DB.CBOR -> Maybe TxCBOR
+    mkTxCBOR = either (const Nothing) (Just . snd) . L.match txCBORPrism
+
+-- | Compute a high level view of a transaction known as 'TransactionInfo'
+-- from a CBOR and a slimmed down version of TxMeta
+mkTransactionInfoFromReadTx :: Monad m
+    => TimeInterpreter m
+    -> W.BlockHeader
+    -> DecoratedTxIns
+    -> EraValue Read.Tx
+    -> SubmissionMeta
+    -> W.TxStatus
+    -> m WT.TransactionInfo
+mkTransactionInfoFromReadTx ti tip decor tx SubmissionMeta{..} status = do
+    txTime <- interpretQuery ti . slotToUTCTime $ submissionMetaSlot
+    return
+        $ WT.TransactionInfo
+        { WT.txInfoId = W.Hash $ txField getEraTxHash
+        , WT.txInfoCBOR = Just $ renderTxToCBOR tx
+        , WT.txInfoFee = txField $ getFee . getEraFee
+        , WT.txInfoInputs = mkTxIn <$> value (getInputs . getEraInputs)
+        , WT.txInfoCollateralInputs = mkTxIn
+            <$> value (getCollateralInputs . getEraCollateralInputs)
+        , WT.txInfoOutputs = value (getOutputs . getEraOutputs)
+        , WT.txInfoCollateralOutput
+            = value (getCollateralOutputs . getEraCollateralOutputs)
+        , WT.txInfoWithdrawals = fold
+            $ value (getWithdrawals . getEraWithdrawals)
+        , WT.txInfoMeta = WT.TxMeta
+              { WT.status = status
+              , WT.direction = submissionMetaDirection
+              , WT.slotNo = submissionMetaSlot
+              , WT.blockHeight = submissionMetaHeight
+              , W.amount = submissionMetaAmount
+              , WT.expiry = fmap (SlotNo . getQuantity . invalidHereafter)
+                    $ value $ getValidity . getEraValidity
+              }
+        , WT.txInfoMetadata = value $ getMetadata . getEraMetadata
+        , WT.txInfoDepth = Quantity $ fromIntegral $
+            if tipH > height
+                  then tipH - height
+                  else 0
+        , WT.txInfoTime = txTime
+        , WT.txInfoScriptValidity
+            = value $ getScriptValidity . getEraScriptValidity
+        }
+  where
+    tipH = getQuantity $ tip ^. #blockHeight
+    height = getQuantity submissionMetaHeight
+
+    txField :: EraFun Read.Tx (K a) -> a
+    txField f = extractEraValue $ applyEraFun f tx
```

Comment:
```suggestion
    txField :: EraFun Read.Tx (K a) -> a
    txField = value
```

Uh — it's the same function? 🤔

Reply:
I discovered a new way of documenting functions :shrug: